server/src/main/java/org/elasticsearch/action/search/SearchResponse.java

```diff
@@ -264,14 +268,23 @@ public Clusters getClusters() {
     }
 
     @Override
-    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
-        builder.startObject();
-        innerToXContent(builder, params);
-        builder.endObject();
-        return builder;
+    public Iterator<? extends ToXContent> toXContentChunked(ToXContent.Params params) {
+        return Iterators.concat(
+            ChunkedToXContentHelper.startObject(),
+            this.innerToXContentChunked(params),
+            ChunkedToXContentHelper.endObject()
+        );
+    }
+
+    public Iterator<? extends ToXContent> innerToXContentChunked(ToXContent.Params params) {
+        return Iterators.concat(
+            Iterators.single((ToXContent) SearchResponse.this::headerToXContent),
+            Iterators.single(clusters),
+            Iterators.flatMap(Iterators.single(internalResponse), r -> r.toXContentChunked(params))
```

Comment:
Similarly I feel there ought to be a nicer way of doing this but I couldn't find one...

Reply:
I think you're looking for this:

```suggestion
            internalResponse.toXContentChunked(params)
```